### PR TITLE
Hunter Stealth Rework

### PR DESCRIPTION
**Pull request by BobSlim**
## About The Pull Request

- Renames some of the hunter defines to be cooler. (variable hierarchy)
- Removes attacking reducing the hunter's invisibility.
- adjust_opacity() is a catch-all to alter the hunter's invisibility. By default, it kicks the hunter out of stealth. It is able to adjust, or set opacity, and then check if stealth still applies.
- taking damage reduces invisibility depending on how much damage is taken. Gives opponents an active way to kick the hunter out of stealth.
- Sneak attack can only occur when below a certain visibility, and with a relatively high cooldown. Should be altered to be similar to the runner's savage in the future.
- Plasma is consumed if you are running or walking. Invisibility increases when you are still or walking, and reduces when you are running.
- If you are stealthed, you do not regain plasma.

## Why It's Good For The Game

Watching the ravenger gave me an idea. The Hunter should be able to fight while partially - or totally invisible. Makes the hunter somewhat similar to the runner, who it succeeds. Gives the hunter some more combat viability, while also not reducing it to "kill one person man".

The hunter in the current meta is absolutely worthless, and not chosen at all. This seeks to remedy that, while also allowing it its invisibility gimmick, and allowing it to use that MORE.

## Changelog
:cl:
add: The Hunter can now fight while invisible! Attacks no longer kick it out of invisibility.
add: The Hunter consumes plasma while moving invisibly, and does not regenerate plasma while stealthed.
add: The Hunter regenerates its invisibility while walking or still, and loses it while running or taking damage.
balance: the Hunter takes longer to ready a sneak attack (10s), and you need to be under a certain level of invisibility.
code: hunter has a catch-all function to alter opacity and kick people outta stealth. (adjust_opacity)
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 1]
```diff
@@ -26,43 +26,20 @@
...
-		COMSIG_XENOMORPH_ATTACK_RAZORWIRE,
-		COMSIG_XENOMORPH_ATTACK_BED,
-		COMSIG_XENOMORPH_ATTACK_NEST,
-		COMSIG_XENOMORPH_ATTACK_TABLE,
-		COMSIG_XENOMORPH_ATTACK_RACK,
-		COMSIG_XENOMORPH_ATTACK_SENTRY,
-		COMSIG_XENOMORPH_ATTACK_M56_POST,
-		COMSIG_XENOMORPH_ATTACK_M56,
-		COMSIG_XENOMORPH_ATTACK_TANK,
 		COMSIG_XENOMORPH_THROW_HIT,
```
Comment (BobSlim) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:29]: shitcode???

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 2]
```diff
@@ -92,85 +69,116 @@
...
+	plasma_mod += 0.0 //do not regenerate plasma while stealthed.
+
+	handle_stealth()
+
+/datum/action/xeno_action/stealth/proc/sneak_attack_zone()
+	if(!stealth || !can_sneak_attack())
+		return
+	return COMSIG_ACCURATE_ZONE
+
+/datum/action/xeno_action/stealth/proc/adjust_opacity(opacity_adjustment = 1, to_set = TRUE) //by default, will cancel stealth.
```
Comment (BobSlim) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:130]: shitcode???

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 3]
```diff
@@ -26,43 +26,20 @@
 	// TODO: attack_alien() overrides are a mess and need a lot of work to make them require parentcalling
 	RegisterSignal(L, list(
 		COMSIG_XENOMORPH_GRAB,
-		COMSIG_XENOMORPH_ATTACK_BARRICADE,
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:29]: removing all these is an issue.

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 4]
```diff
@@ -26,43 +26,20 @@
...
+		SIGNAL_ADDTRAIT(TRAIT_KNOCKEDOUT), 
+		SIGNAL_ADDTRAIT(TRAIT_FLOORED)), .proc/adjust_opacity)
 
 	RegisterSignal(src, COMSIG_XENOMORPH_TAKING_DAMAGE, .proc/damage_taken)
 
 /datum/action/xeno_action/stealth/remove_action(mob/living/L)
 	UnregisterSignal(L, list(
 		COMSIG_XENOMORPH_POUNCE, 
 		COMSIG_XENO_LIVING_THROW_HIT, 
-		COMSIG_XENOMORPH_ATTACK_LIVING,
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:52]: this specific signal still needs to be unregistered

**Pull request closed without merging**